include/SDL3/SDL_gpu.h

```diff
@@ -0,0 +1,2251 @@
+    SDL_bool stencilTestEnable;
+    SDL_GpuStencilOpState backStencilState;
+    SDL_GpuStencilOpState frontStencilState;
+    Uint32 compareMask;
+    Uint32 writeMask;
+    Uint32 reference;
+} SDL_GpuDepthStencilState;
+
+typedef struct SDL_GpuColorAttachmentDescription
+{
+    SDL_GpuTextureFormat format;
+    SDL_GpuColorAttachmentBlendState blendState;
+} SDL_GpuColorAttachmentDescription;
+
+typedef struct SDL_GpuGraphicsPipelineAttachmentInfo
+{
+    SDL_GpuColorAttachmentDescription *colorAttachmentDescriptions;
+    Uint32 colorAttachmentCount;
+    SDL_bool hasDepthStencilAttachment;
+    SDL_GpuTextureFormat depthStencilFormat;
+} SDL_GpuGraphicsPipelineAttachmentInfo;
+
+typedef struct SDL_GpuGraphicsPipelineCreateInfo
+{
+    SDL_GpuShader *vertexShader;
+    SDL_GpuShader *fragmentShader;
+    SDL_GpuVertexInputState vertexInputState;
+    SDL_GpuPrimitiveType primitiveType;
+    SDL_GpuRasterizerState rasterizerState;
+    SDL_GpuMultisampleState multisampleState;
+    SDL_GpuDepthStencilState depthStencilState;
+    SDL_GpuGraphicsPipelineAttachmentInfo attachmentInfo;
+    float blendConstants[4];
+} SDL_GpuGraphicsPipelineCreateInfo;
+
+typedef struct SDL_GpuComputePipelineCreateInfo
+{
+    size_t codeSize;
+    const Uint8 *code;
+    const char *entryPointName;
+    SDL_GpuShaderFormat format;
+    Uint32 readOnlyStorageTextureCount;
+    Uint32 readOnlyStorageBufferCount;
+    Uint32 readWriteStorageTextureCount;
+    Uint32 readWriteStorageBufferCount;
+    Uint32 uniformBufferCount;
+    Uint32 threadCountX;
+    Uint32 threadCountY;
+    Uint32 threadCountZ;
+} SDL_GpuComputePipelineCreateInfo;
+
+typedef struct SDL_GpuColorAttachmentInfo
+{
+    /* The texture slice that will be used as a color attachment by a render pass. */
+    SDL_GpuTextureSlice textureSlice;
+
+    /* Can be ignored by RenderPass if CLEAR is not used */
+    SDL_FColor clearColor;
+
+    /* Determines what is done with the texture slice at the beginning of the render pass.
+     *
+     *   LOAD:
+     *     Loads the data currently in the texture slice.
+     *
+     *   CLEAR:
+     *     Clears the texture slice to a single color.
+     *
+     *   DONT_CARE:
+     *     The driver will do whatever it wants with the texture slice memory.
+     *     This is a good option if you know that every single pixel will be touched in the render pass.
+     */
+    SDL_GpuLoadOp loadOp;
+
+    /* Determines what is done with the texture slice at the end of the render pass.
+     *
+     *   STORE:
+     *     Stores the results of the render pass in the texture slice.
+     *
+     *   DONT_CARE:
+     *     The driver will do whatever it wants with the texture slice memory.
+     *     This is often a good option for depth/stencil textures.
+     */
+    SDL_GpuStoreOp storeOp;
+
+    /* if SDL_TRUE, cycles the texture if the texture slice is bound and loadOp is not LOAD */
+    SDL_bool cycle;
+} SDL_GpuColorAttachmentInfo;
```

Comment:
Does it make any sense to add `SDL_PropertiesID props;` to any of these structures for future expansion?

Reply:
Tracking at https://github.com/thatcosmonaut/SDL/issues/157